### PR TITLE
[#1206] Frequent call of update chain tip

### LINKED ISSUE #1206: Frequent call of update chain tip
When syncing (Spend before Sync strategy) takes a long time the error of expiry height may happen. See the [librustzcash ticket ](https://github.com/zcash/librustzcash/issues/906) for more information. 

We discussed how to approach it and the solution is to call `updateChainTip` in repeatedly. When the wallet is synced this is covered with 20-30s automatic refreshing mechanism. When the wallet is restored or will need to sync a huge amount of blocks, there's a big chance the Spend before Sync won't be called again during 10+ minutes. 

We need exactly these cases to be covered as well. The solution is to call `updateChainTip` from time to time. The proposal is to incorporate it to the loop of scan ranges processing. Once the check passes, the State Machine takes a special route to update the chain tip and then continue with the processing. This way expiry height errors can be fully avoided.

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #1206 

- The State Machine has been slightly updated so it measures time when it lastly updated chain tip. If it happened more than 10mins ago, it calls the .updateChainTip action once again before the download-scan-enhance loop continues
- updated unit test

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: Sources/ZcashLightClientKit/Block/Actions/UpdateChainTipAction.swift [hunk 1]
```diff
@@ -17,19 +17,40 @@ final class UpdateChainTipAction {
...
+
+            await context.update(state: .processSuggestedScanRanges)
+        } else if await context.prevState == .enhance {
+            // Case B - called from the `download-scan-enhance` loop so the chain tip is updated at least once in 10 minutes
+            let lastChainTipUpdateTime = await context.lastChainTipUpdateTime
+            let now = Date().timeIntervalSince1970
 
+            // when the diff is more than 10 minutes, do the update
+            if now - lastChainTipUpdateTime > 600 {
+                try await updateChainTip(context, time: now)
```
Comment (nuttycom) [inline Sources/ZcashLightClientKit/Block/Actions/UpdateChainTipAction.swift:48]: Will this trigger an update of the suggested scan ranges that are being used in the download-scan-enhance loop? If not, it needs to do so; essentially every time `updateChainTip` is called the scan ranges should be reevaluated.

Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Block/Actions/UpdateChainTipAction.swift:48]: No it won't, thanks for bringing it up. For now it simply continues, so I'll update this PR in the morning.

Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Block/Actions/UpdateChainTipAction.swift]: Now it starts the chain tip update + suggest scan ranges every 10 minutes

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**